Include/cpython/abstract.h

```diff
@@ -50,7 +50,8 @@ PyAPI_FUNC(PyObject *) _PyObject_MakeTpCall(
     PyObject *const *args, Py_ssize_t nargs,
     PyObject *keywords);
 
-#define PY_VECTORCALL_ARGUMENTS_OFFSET ((size_t)1 << (8 * sizeof(size_t) - 1))
+#define PY_VECTORCALL_ARGUMENTS_OFFSET \
+    (_Py_static_cast(size_t, 1) << (8 * sizeof(size_t) - 1))
```

Comment:
This macro should be updated since it's used in a static inline function: PyVectorcall_NARGS().

It seems like the warning is emitted if an old-style cast is done in a static inline function, but it seems to be fine if it's only done in macros.